### DISCUSSION TITLE
registered namespace error

### DISCUSSION THREAD

**Discussion by jakeymac** [tethysplatform/tethys#975] [Q&A]
I'm just getting started with Tethys. I installed everything yesterday and began loading up a new project application, and after a bit of time, I started getting this error when logging in the Tethys Portal/opening apps.

File "/Users/quest/miniconda3/envs/tethys/lib/python3.11/site-packages/django/urls/base.py", line 82, in reverse
    raise NoReverseMatch("%s is not a registered namespace" % key)

I've tried uninstalling the app and even Tethys itself and reinstalling, and am getting the same error again. 

Any tips for this issue?
@swainn

**Comment by jakeymac** [comment]
It seems to be giving me an issue with this line in app_card.html 
href="{% if app.proxied %}{{ app.url }}{% if app.back_url %}?back={{ app.back_url }}{% endif %}{% else %}{% url app.index_url %}{% endif %}"

**Comment by jakeymac** [comment]
specifically in the url app.index_url tag

**Comment by swainn** [comment]
Hey @jakeymac can you send me the full Traceback you are getting?

Two things to check:

1. Did you make any changes to any of these parameters in the app class in app.py: `package`, `index`, `root_url`. Those are sort of sensitive parameters and shouldn't be changed unless you know what you are doing.
2. Look for syntax errors, import errors, missing imports, typos, etc. in `controllers.py` and `app.py`, or any module in your code imported by those files. Any of those can prevent the app from loading and sometimes even prevent it from being installed/uninstalled, unfortunately.

Worst comes to worst, you can uninstall using pip and start over:

```
pip uninstall tethysapp-<app_name>
```
